### PR TITLE
Adding search query and current date header row to CSV export

### PR DESCRIPTION
**Pull request by pjsier**
Added the first row to the CSV described in #306. Includes the current date, and then a list of all filters from the advanced search (ignoring null/empty values) or just the search query. Example output is `""Created: 2017-08-01 - Query: search_model = participant, event_tags = swimming,dia_muertos"`

### CODE REVIEW

File: streetcrm/admin.py [hunk 1]
```diff
@@ -539,7 +539,6 @@ def advanced_search_do(self, request, form, export):
         # Remove any without participants
         # results = self._remove_empty_values(results)
 
-
```
Comment (kfogel) [inline streetcrm/admin.py]: Not a big deal, but it's good to avoid unrelated whitespace changes -- they are just a distraction in the diff, since the reviewer will assume that any visible textual change is part of the semantic change (and then discover otherwise).

File: streetcrm/admin.py [hunk 2]
```diff
@@ -552,8 +551,24 @@ def advanced_search_do(self, request, form, export):
...
+                elif hasattr(value, '__iter__') and len(value):
+                    search_params.append('{} = {}'.format(key, ','.join([str(v) for v in value])))
+                elif hasattr(value, 'pk'):
+                    search_params.append('{} = {}'.format(key, str(value)))
+
+            search_header = ', '.join(search_params)
+
             results_package = {"form": form,
                                "search_results": results,
+                               "search_header": search_header,
```
Comment (kfogel) [inline streetcrm/admin.py]: I'm really passing on a comment from @cecilia-donnelly here, expanded slightly by me: wouldn't `export_search` be the right place to assemble this new search_header?  IOW, instead of assembling it in `advanced_search_do` (which doesn't know whether or not its results will ever be exported), do it in `export_search`, where by definition we know that the export is happening.

File: streetcrm/admin.py [hunk 3]
```diff
@@ -638,15 +655,18 @@ def export_search(request, form, advanced, search_query):
         
         last_header=[]
         header=[]
+        
+        search_title = 'Created: {} - Query: {}'.format(
```
Comment (kfogel) [inline streetcrm/admin.py]: We might change the exact wording here, but that's TBD.  Obviously, it's a trivial change.

File: streetcrm/admin.py (hunk 2 above)
Comment (cecilia-donnelly) [inline streetcrm/admin.py]: Right, exactly.  Thanks, @kfogel.  

Now that I've had a chance to look at it more closely, it might be difficult to assemble in `export_search`, since we don't pass `data` to `export_search`, so we wouldn't be able to use the `for` loop approach.  We could abstract that into a different function, called from `advanced_search_do`, to assemble the query parameters into a string.  

What do you think about naming the parameters something other than their keys in the form data?  For example, `search_model` is a little unintuitive to the everyday user.  That might make this more complicated than we want, but I'm curious about your thoughts.

Comment (kfogel) [inline streetcrm/admin.py]: A better solution might be to keep the correspondence between the form keys and parameters listed in the CSV header row (because that's easy to maintain & is intuitive for other developers encountering the code), but just give the form keys better names, where "better" means "more meaningful to an end user".

Comment (kfogel) [inline streetcrm/admin.py]: @pjsier thoughts? :-)

Comment (pjsier) [inline streetcrm/admin.py]: Thanks for the review! I think renaming the form keys is probably the easiest option for maintainability. In that case, would it still be worth splitting the search string creation into a separate method?

File: streetcrm/admin.py (hunk 3 above)
Comment (pjsier) [inline streetcrm/admin.py]: Yep, open to changes there, and also want to call out that the only way I'm differentiating tags from other comma separated items is not including a space between them, so that could definitely be improved.

File: streetcrm/admin.py (hunk 2 above)
Comment (kfogel) [inline streetcrm/admin.py]: Personally I don't have a strong opinion on whether a separate method is called for -- really, could go either way.  So unless @cecilia-donnelly does have a strong opinion, I'd say just do whichever way feels most right (probably easier to know once you see the updated code in place anyway).

Comment (pjsier) [inline streetcrm/admin.py]: Just updated the `search_model` key to be `search_for`. The only other one I was one the fence about was `leader_stage`, but it looks like that phrasing is used in other user-facing places. Happy to any other keys that look off though.

**Comment by cecilia-donnelly** [comment]
I made a small change to the header text, as you can see.  Merging now!
